src/main/java/uk/nhs/tis/trainee/notifications/service/PlacementService.java

```diff
@@ -137,7 +159,9 @@ private Map<NotificationType, Instant> getNotificationsSent(String traineeId,
   public void addNotifications(Placement placement)
       throws SchedulerException {
 
-    deleteNotifications(placement); //first delete any stale notifications
+    //first delete any stale notifications
+    deleteNotifications(placement);
+    deleteScheduledInAppNotifications(placement.getPersonId());
```

Comment:
I don't think we can delete _all_ scheduled notifications for the trainee, only those for the specific placement we have just received.